src/behaviours/cluster/kube-object-status-icon/show-status-for-a-kube-object.test.tsx

```diff
@@ -0,0 +1,376 @@
+                {tooltip.children || tooltip}
+              </div>
+            </>
+          );
+        }
+
+        return <Target {...props} />;
+      },
+}));
+
+describe("show status for a kube object", () => {
+  let builder: ApplicationBuilder;
+  let infoStatusIsShown: boolean;
+  let warningStatusIsShown: boolean;
+  let criticalStatusIsShown: boolean;
+
+  beforeEach(() => {
+    useFakeTime("2015-10-21T07:28:00Z");
+
+    builder = getApplicationBuilder();
+
+    const rendererDi = builder.dis.rendererDi;
+
+    infoStatusIsShown = false;
+
+    const infoStatusInjectable = getInjectable({
+      id: "some-info-status",
+      injectionToken: kubeObjectStatusTextInjectionToken,
+
+      instantiate: () => ({
+        apiVersions: ["some-api-version"],
+        kind: "some-kind",
+        enabled: computed(() => true),
+
+        resolve: (resource) => infoStatusIsShown ? ({
+          level: KubeObjectStatusLevel.INFO,
+          text: `Some info status for ${resource.getName()}`,
+          timestamp: "2015-10-19T07:28:00Z",
+        }) : null,
+      }),
+    });
+
+    warningStatusIsShown = false;
+
+    const warningStatusInjectable = getInjectable({
+      id: "some-warning-status",
+      injectionToken: kubeObjectStatusTextInjectionToken,
+
+      instantiate: () => ({
+        apiVersions: ["some-api-version"],
+        kind: "some-kind",
+        enabled: computed(() => true),
+
+        resolve: (resource) => warningStatusIsShown ? ({
+          level: KubeObjectStatusLevel.WARNING,
+          text: `Some warning status for ${resource.getName()}`,
+          timestamp: "2015-10-19T07:28:00Z",
+        }) : null,
+      }),
+    });
+
+    criticalStatusIsShown = false;
+
+    const criticalStatusInjectable = getInjectable({
+      id: "some-critical-status",
+      injectionToken: kubeObjectStatusTextInjectionToken,
+
+      instantiate: () => ({
+        apiVersions: ["some-api-version"],
+        kind: "some-kind",
+        enabled: computed(() => true),
+
+        resolve: (resource) => {
+          return criticalStatusIsShown
+            ? {
+              level: KubeObjectStatusLevel.CRITICAL,
+              text: `Some critical status for ${resource.getName()}`,
+              timestamp: "2015-10-19T07:28:00Z",
+            }
+            : null;
+        },
+      }),
+    });
+
+    rendererDi.register(
+      testRouteInjectable,
+      testRouteComponentInjectable,
+      infoStatusInjectable,
+      warningStatusInjectable,
+      criticalStatusInjectable,
+      someAtomInjectable,
+    );
+
+    builder.setEnvironmentToClusterFrame();
+  });
+
+  describe("given application starts and in test page", () => {
+    let rendererDi: DiContainer;
+    let rendered: RenderResult;
+    let rerenderParent: () => void;
+
+    beforeEach(async () => {
+      rendered = await builder.render();
+
+      rendererDi = builder.dis.rendererDi;
+
+      const someAtom = rendererDi.inject(someAtomInjectable);
```

Comment:
Would be good to know better what atom is and why it it needed in tests.

Reply:
`Atom` is concept of `mobx`. Documentation here: https://mobx.js.org/custom-observables.html
Basically it's something like `observable` or `computed` but without any value. There's just a way to say:
- `atom.reportObserved()` which means that you are interested of it, e.g. component re-rendering is controlled by it (like when you call `someObservable.get()` in component.)
- `atom.reportChanged()` which means that every place with `reportObserved()` should be recomputed (like when you call `someObservable.set("some-value")`)

In this test I'm using it to control when parent component should be re-rendered. 
I could do the same with `observable.box` which I would dereference (`.get()`) in the component and `.set("some-value")` when I want to re-render. Why Atom? - because it doesn't require the value which changes.